src/components/Banner/AnonBannerSticky.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { DesktopOnly, MobileOnly } from './../Responsive'
+import BannerDesktop from './BannerDesktop'
+import BannerMobile from './BannerMobile'
+import StickyBannerContent from './StickyBannerContent'
+
+class AnonBannerSticky extends React.PureComponent {
+  state = { isVisible: false }
+
+  componentDidMount () {
+    this.viewportHeight = window.innerHeight
+  }
+
+  componentDidUpdate (prevProps) {
+    if (!prevProps.bannerStaticRef && this.props.bannerStaticRef) {
+      this.checkVisibility()
+      this.timer = setTimeout(this.checkVisibility, 50)
+    }
+  }
```

Comment:
timer - small hack for new check after short period. I did it, because images in insights can loading, when banner has already mounted. This timer for them. I know, that when slow network it can be so short period, but onLoad() for image I can't use, because I didn't see in Editor this ability

Reply:
@DmitriVanGuard @Partysun what do you think?

Reply:
There is no need for it.
As you pointed out, `50` milliseconds is too fast for slow connections. It's even fast for `4g`. In result, it gives us 0 benefits and 2 extra complexities: lifecycle method and timer.

Reply:
ok, but if you open long insight with image and won't scroll - you won't see any banner, because data of top static banner will be incorrect. 150 - ok for 4g, I can change

Reply:
If it's a long insight user will scroll to see its content which will result in a banner popping up. This arbitrary value does not guaranty anything.

Reply:
ok, I'll remove it